### ISSUE TITLE
where() with empty function

### ISSUE THREAD

**Issue by whitelynx** [tgriesser/knex#749]
Similar to #477, if you pass `.where()` an empty function (or one where no logic path causes it to call `this.where()`), it generates invalid SQL:
```javascript
knex().select('foo').from('tbl').where(function() {});
```
...generates:
```sql
select "foo" from "tbl" where ()
```

It should either simply leave off the `where` clause, or do something similar to `where true`. (or whatever is required by sqlite)

**Comment by radio-miskovice** [comment]
I vote for this bug/feature. I have the same problem in my app. If I have no condition to apply, I must test beforehand and then choose to execute one of two almost identical queries (same excpet the where() call).  This makes programming using knex unnaturally complicated and it is then worth not to use knex at all.

**Comment by tgriesser** [comment]
Fixed/tested in 0.8
